src/pykeen/nn/modules.py

```diff
@@ -1413,6 +1416,64 @@ def _prepare_hrt_for_functional(
         return dict(h=h, r=r, c_r=c_r, t=t)
 
 
+class BoxEInteraction(Interaction):
+    """An implementation of the BoxE interaction from [abboud2020]_."""
+
+    relation_shape = ("d", "d", "s", "d", "d", "s")  # Boxes are 2xd (size) each, x 2 sets of boxes: head and tail
+    entity_shape = ("d", "d")  # Base position and bump
+
+    def __init__(self, tanh_map: bool = True, norm_order: int = 2):
+        r"""
+        Instantiate the interaction module.
+
+        :param tanh_map:
+            Should the hyperbolic tangent be applied to all representations prior to model scoring?
+        :param norm_order:
+            The normalization order (default 2)
+
+        .. math::
+            points: p
+            box_lows: l
+            box_highs: h
+
+            w = h - l . Width is the difference between the upper and lower box bound
+            c = (h + l) / 2. Box centers (the mean of the box bounds)
+
+            dist(p,l,h) = |p-c|/(w+1) if l <= p <+ h, |p-c|*(w+1) - 0.5*w*((w+1)-1/(w+1)) otherwise.
+        """
+        super().__init__()
+        self.tanh_map = tanh_map  # Map the tanh map
+        self.norm_order = norm_order
+
+    def forward(self, h, r, t):
+        rh_base, rh_delta, rh_size, rt_base, rt_delta, rt_size = r
+        h_pos, h_bump = h
+        t_pos, t_bump = t
+        # First, compute the boxes
+        rh_low, rh_high = compute_box(rh_base, rh_delta, rh_size)
+        rt_low, rt_high = compute_box(rt_base, rt_delta, rt_size)
+
+        score_h = boxe_kg_arity_position_computation(
+            entity_pos=h_pos,
+            other_entity_bump=t_bump,
+            relation_box_low=rh_low,
+            relation_box_high=rh_high,
+            tanh_map=self.tanh_map,
+            norm_order=self.norm_order,
+        )
+
+        score_t = boxe_kg_arity_position_computation(
+            entity_pos=t_pos,
+            other_entity_bump=h_bump,
+            relation_box_low=rt_low,
+            relation_box_high=rt_high,
+            tanh_map=self.tanh_map,
+            norm_order=self.norm_order,
+        )
+        total_score = score_h + score_t
+        return -total_score  # Because this is inverted in NSSALoss (higher is better)
```

Comment:
In general, we follow the convention that larger scores indicate more likely triples. Hence, for distance based models, we usually use `score = -distance` (although of course other anti-monotonous transformation are also possible).

Reply:
Can you make this shorter by just doing `score_t - score_h` and adding a comment that you flipped it to make it negative

Reply:
>Can you make this shorter by just doing score_t - score_h and adding a comment that you flipped it to make it negative. Then you save one operation

would be `-score_h - score_t`, right? In that case, no operation was saved

Reply:
Yes. You are right. I have now lost the skill to do basic algebra. At first I just couldn't do arithmetic. Soon I will forget calculus as well, I guess